twine/distribution.py

```diff
@@ -0,0 +1,8 @@
+class Distribution:
```

Comment:
Nitpick, nonblocking: maybe this should be an ABC or an interface, assuming it isn't intended to be instantiated directly?

Reply:
What do you mean with interface? A `typing.Protocol`? Protocols are useful when direct inheritance is impossible or not convenient to implement. Here we have control of all subclasses, thus I don't see a protocol adding any value. An ABC would be more appropriate, but for a simple case like this I see it as adding more noise than value. I can use an ABC if there strong desire for it.